### PR TITLE
Ensure pressing 'Enter' in SelectPanel filter input causes navigation if first item is link

### PR DESCRIPTION
**Pull request by camertron**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

In a previous PR, I added functionality to ensure that pressing <kbd>Enter</kbd> while focused on the filter input of a `SelectPanel` selects the first list item, in accordance with our API docs for the component. However, the methodology I chose did not account for the case when list items are links, i.e. anchor elements. This PR simulates a `click` event instead, so everything works as expected even when list items are links.

### Screenshots

The video below shows that pressing <kbd>Enter</kbd> while focused on the filter input navigates to the URL specified in the anchor tag for the first list item, which is https://github.com.

https://github.com/user-attachments/assets/da4bc099-8c0f-4d54-bc0f-3ecab58cf437

### Integration

No changes necessary in production.

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### Merge checklist

- [x] Added/updated tests
~- [ ] Added/updated documentation~
- [x] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by kendallgassner** [comment]
I tested in Lookbook and discovered a niche bug. In the single select playground example, if you select an item via the input, then filter the input by some text and select a different item, both items are selected when you reopen the panel.

**Comment by kendallgassner** [comment]
Still seeing the same issue: 

https://github.com/user-attachments/assets/4c36ea05-5033-4cc7-8b26-6caff8baa935

Alt: On storybook `SelectPanel/playground` I navigate to an item using the search field. I check that item. When I reopen the SelectPanel two items are selected even though I am in a SingleSelect.

File: app/components/primer/alpha/select_panel_element.ts [hunk 1]
```diff
@@ -84,6 +84,7 @@ export class SelectPanelElement extends HTMLElement {
   #selectedItems: Map<string, SelectedItem> = new Map()
   #loadingDelayTimeoutId: number | null = null
   #loadingAnnouncementTimeoutId: number | null = null
+  #userHasMadeSelection: boolean = false
```
Comment (kendallgassner) [inline app/components/primer/alpha/select_panel_element.ts:87]: For multiselect, If s server returns a subset of items (say 10 out of 100 and all are selected) then when we filter by input and get a new set of items some items will not be selected that should be selected.

Comment (camertron) [reply app/components/primer/alpha/select_panel_element.ts:87]: Good catch!

**Comment by camertron** [comment]
Alright, we ended up removing the piggybacking code designed to fix issues where a user's selections would be overridden by server-rendered selections. This PR now only contains fixes for the issue in the title + description.

**Pull request merged**